flink-clients/src/main/java/org/apache/flink/client/program/Client.java

```diff
@@ -277,12 +278,20 @@ else if (prog.isUsingInteractiveMode()) {
 			// invoke here
 			try {
 				prog.invokeInteractiveModeForExecution();
+				// if we're here, it means the program is valid for detached mode execution.
+				JobWithJars plan = ContextEnvironment.ContextEnvironmentFactory.getPlan();
+				JobGraph graph = ContextEnvironment.ContextEnvironmentFactory.getGraph();
+				if (plan != null && graph == null) {
```

Comment:
What would happen if we created a streaming program and a batch program in one jar? Then both would be set and only the batch program would be created...